Datastore/src/Operation.php

```diff
@@ -614,12 +689,20 @@ public function commit(array $mutations, array $options = [])
             'databaseId' => $this->databaseId,
         ];
 
-        $res = $this->connection->commit($options + [
-            'mode' => ($options['transaction']) ? 'TRANSACTIONAL' : 'NON_TRANSACTIONAL',
-            'mutations' => $mutations,
-            'projectId' => $this->projectId,
-        ]);
+        $options = $this->parseCommitOptions($mutations, $options);
+
+        list($data, $optionalArgs) = $this->splitOptionalArgs($options);
```

Comment:
It would make more sense to swap these - since we know that none of the options we add in `parseCommitOptions` will be part of `$optionalArgs`, and we know that none of `$optionalArgs` should be used in of `parseCommitOptions`. 

```suggestion
        list($data, $optionalArgs) = $this->splitOptionalArgs($options);
        $data = $this->parseCommitOptions($mutations, $data);
```